sdk/state/close_test.go

```diff
@@ -107,16 +109,41 @@ func TestChannel_ProposeClose(t *testing.T) {
 		MaxOpenExpiry:       2 * time.Hour,
 	})
 
-	open1, err := localChannel.ProposeOpen(OpenParams{
-		ObservationPeriodTime:      1,
-		ObservationPeriodLedgerGap: 1,
-		ExpiresAt:                  time.Now().Add(time.Hour),
-	})
-	require.NoError(t, err)
-	open2, err := remoteChannel.ConfirmOpen(open1)
-	require.NoError(t, err)
-	_, err = localChannel.ConfirmOpen(open2)
-	require.NoError(t, err)
+	// Put channel into the Open state.
+	{
+		open1, err := localChannel.ProposeOpen(OpenParams{
+			ObservationPeriodTime:      1,
+			ObservationPeriodLedgerGap: 1,
+			ExpiresAt:                  time.Now().Add(time.Hour),
+		})
+		require.NoError(t, err)
+		open2, err := remoteChannel.ConfirmOpen(open1)
+		require.NoError(t, err)
+		_, err = localChannel.ConfirmOpen(open2)
+		require.NoError(t, err)
+
+		ftx, err := localChannel.OpenTx()
+		require.NoError(t, err)
+		ftxXDR, err := ftx.Base64()
+		require.NoError(t, err)
+
+		successResultXDR, err := txbuildtest.BuildResultXDR(true)
+		require.NoError(t, err)
+		resultMetaXDR, err := txbuildtest.BuildFormationResultMetaXDR(txbuildtest.FormationResultMetaParams{
+			InitiatorSigner: localSigner.Address(),
+			ResponderSigner: remoteSigner.Address(),
+			InitiatorEscrow: localEscrowAccount.Address.Address(),
+			ResponderEscrow: remoteEscrowAccount.Address.Address(),
+			StartSequence:   localEscrowAccount.SequenceNumber + 1,
+			Asset:           txnbuild.NativeAsset{},
+		})
+		require.NoError(t, err)
+
+		err = localChannel.IngestTx(ftxXDR, successResultXDR, resultMetaXDR)
+		require.NoError(t, err)
+		err = remoteChannel.IngestTx(ftxXDR, successResultXDR, resultMetaXDR)
+		require.NoError(t, err)
+	}
```

Comment:
:bulb: The improvements to the tests look great. The fact that so many places in the tests needed updating for the state to be accurate, I think it would be worth adding into every test that was changed assertions on the result of calling `channel.State()`, that way these tests will ensure that the state value returned from the function is consistent over time. Wdyt?